### DISCUSSION TITLE
Incorrect Control Bounds data

### DISCUSSION THREAD

**Discussion by markusalbrecht-procam** [AvaloniaUI/Avalonia#14392] [General]
Hi there,
we are playing with this since several days now as we encountered problems getting the 3D control Viewport fully sized via the Bounds. In order to find the source of the issue we removed almost any view data and only kept the control full sized to the window. The result is always wrong --> delivering 30% less then the actual max window width and height:
![image](https://github.com/AvaloniaUI/Avalonia/assets/56921919/1dbc727a-078c-4521-8d54-63842897276a)
Should be at least 1900*980 when looking at the real window size:
![image](https://github.com/AvaloniaUI/Avalonia/assets/56921919/ccca9dcd-8ae1-4067-9afd-4ac69f4b8c65)
So it seems that the Bounds calculation of Avalonia is doing something but not getting the real size of the control or window for some reason ? 
![image](https://github.com/AvaloniaUI/Avalonia/assets/56921919/cf35a9bc-dce8-4cb7-a217-ff7a0149eacb)
What is the problem here ?
![image](https://github.com/AvaloniaUI/Avalonia/assets/56921919/9edf0365-c563-4c4b-b856-36ad0e6a0010)

**Comment by markusalbrecht-procam** [comment]
By adding a factor of 1.25 it works but I do not think that this can be the solution as it might differ from GPU to GPU and Res to Res ?!?!
![image](https://github.com/AvaloniaUI/Avalonia/assets/56921919/339f3ffa-8b50-497b-bc9c-fe81cd9ae6e7)

**Comment by stevemonaco** [comment]
Even when setting the system scaling here to 1.25x (Windows 11), I'm still getting the correct Bounds in Avalonia 11.0.6 for the top-level `<Window>` when inspecting via DevTools (F12). Some more information as to which OS you're using (looks like Win), Avalonia version, and which control you're querying for Bounds might help troubleshoot.

**Reply by markusalbrecht-procam** [reply]
This solved the issue 
![image](https://github.com/AvaloniaUI/Avalonia/assets/56921919/0925cb34-7ebc-4554-8717-0aa6893ef936)

**Comment by maxkatz6** [comment]
Render scaling. You can get your render scaling from TopLevel/Window.RenderScaling property.

**Reply by markusalbrecht-procam** [reply]
@maxkatz6 Thanks this solved the issue right away ;)

**Reply by anose001** [reply]
@maxkatz6, I tried the above approach via TopLevel on Android in the MainView.axaml.cs file but received a null reference. Is there a proper way of retrieveing RenderScaling or dpi values on mobile?
